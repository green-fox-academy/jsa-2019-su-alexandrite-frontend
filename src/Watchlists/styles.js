import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  watchListTitle: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  expandButton: {
    padding: 15,
    marginVertical: -15,
    marginRight: -15,
  },
  watchListItem: {
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    borderBottomColor: '#eee',
    borderBottomWidth: 1,
    padding: 10,
  },
  watchListItemColumn: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  watchListItemColumnText: {
    fontSize: 12,
    color: '#666',
  },
  watchListCardTitle: {
    fontSize: 13,
    lineHeight: 18,
    fontWeight: '600',
    color: '#333',
  },
  addIcon: {
    right: 0,
    bottom: 0,
    position: 'absolute',
  },
  watchlistField: {
    padding: 8,
    borderBottomWidth: 1,
    borderBottomColor: '#eee',
  },
  editorMode: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: 10,
  },
  editorModeLeft: {
    flex: 1,
    flexDirection: 'row',
  },
  editorButton: {
    paddingHorizontal: 10,
    paddingVertical: 8,
    borderRadius: 6,
  },
  editorModeRight: {
    flexDirection: 'row',
  },
  editor: {
    flex: 1,
    alignSelf: 'flex-end',
    marginTop: 10,
  },
  checkButton: {
    fontSize: 16,
    marginVertical: -5,
    marginRight: 25,
  },
  trashIcon: {
    paddingBottom: 0,
  },
  editDeleteButton: {
    color: '#b40d0d',
    fontWeight: 'bold',
    fontSize: 12,
    textAlignVertical: 'center',
  },
  detailIcon: {
    padding: 10,
    margin: -10,
  },
});
