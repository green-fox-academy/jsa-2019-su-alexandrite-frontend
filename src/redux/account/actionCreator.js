import { LOCAL_URL } from 'react-native-dotenv';
import {
  LOGIN_USER_FAILURE,
  LOGIN_USER_SUCCESS,
  LOGOUT_SUCCESS,
} from './actionType';

const loginUserSuccess = (payload) => ({
  type: LOGIN_USER_SUCCESS,
  payload,
});

export const loginUserFail = (payload) => ({
  type: LOGIN_USER_FAILURE,
  payload,
});

const loginUser = (username, password) => (dispatch) => {
  const loginURL = `${LOCAL_URL}/users/login`;
  fetch(loginURL, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      username,
      password,
    }),
  })
    .then((response) => {
      if (response.status === 200) {
        return response.json();
      }
      throw new Error('Oops! there is something wrong with your app');
    })
    .then((response) => {
      dispatch(loginUserSuccess(response));
    })
    .catch((error) => dispatch(loginUserFail(error)));
};

export const logOut = () => ({
  type: LOGOUT_SUCCESS,
});

export default loginUser;
