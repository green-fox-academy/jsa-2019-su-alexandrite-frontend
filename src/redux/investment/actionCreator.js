import { SERVER_URL, API_URL, API_KEY } from 'react-native-dotenv';
import {
  FETCH_PORTFOLIO_DETAILS_START,
  FETCH_PORTFOLIO_DETAILS_FAIL,
  FETCH_PORTFOLIO_DETAILS_SUCCESS,
} from './actionType';
import { round, moneyAmount2String } from '../../common/numbers';

export const fetchPortfolioDetailsStart = () => ({
  type: FETCH_PORTFOLIO_DETAILS_START,
});

export const fetchPortfolioDetailsFail = (payload) => ({
  type: FETCH_PORTFOLIO_DETAILS_FAIL,
  payload,
});

export const fetchPortfolioDetailsSuccess = (payload) => ({
  type: FETCH_PORTFOLIO_DETAILS_SUCCESS,
  payload,
});

export const calculatePortfolioValue = () => (dispatch, getState) => {
  const serverUrl = new URL(`${SERVER_URL}/investments/user/`);
  const headers = new Headers();
  const { accessToken } = getState().user;
  headers.append('authorization', `Bearer ${accessToken}`);
  dispatch(fetchPortfolioDetailsStart());
  fetch(serverUrl, { headers })
    .then((response) => {
      if (!response.ok) {
        switch (response.status) {
          case 401:
            throw new Error('Sorry, we cannot validate your identity. Please login and try again.');
          default:
            throw new Error('Oops, there\'s something wrong with our app.');
        }
      }
      return response.json();
    })
    .then((stocks) => {
      const symbols = stocks.map((stock) => stock.symbol);
      const apiUrl = new URL(`${API_URL}/stock/market/batch`);
      apiUrl.searchParams.append('symbols', symbols);
      apiUrl.searchParams.append('types', 'price,company,logo');
      apiUrl.searchParams.append('token', API_KEY);
      fetch(apiUrl)
        .then((res) => {
          if (!res.ok) {
            throw new Error('Oops, there\'s something wrong with our app.');
          }
          return res.json();
        })
        .then((res) => {
          const instruments = {};
          let totalValue = stocks
            .map(({ shares, symbol }) => {
              instruments[symbol] = {
                logo: res[symbol].logo.url,
                company: res[symbol].company.companyName,
                description: res[symbol].company.description,
                marketValue: round(shares * res[symbol].price),
              };
              return shares * res[symbol].price;
            })
            .reduce((a, b) => a + b);
          totalValue = moneyAmount2String((totalValue));
          dispatch(fetchPortfolioDetailsSuccess({
            totalValue,
            stocks,
            instruments,
          }));
        });
    })
    .catch((err) => dispatch(fetchPortfolioDetailsFail(err)));
};
