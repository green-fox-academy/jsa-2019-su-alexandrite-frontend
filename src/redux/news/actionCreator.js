import {
  SERVER_URL,
} from 'react-native-dotenv';
import {
  FETCH_INVESTED_STOCKS_START,
  FETCH_INVESTED_STOCKS_FAIL,
  FETCH_INVESTED_STOCKS_SUCCESS,
} from './actionType';

export const fetchInvestedStocksStart = () => ({
  type: FETCH_INVESTED_STOCKS_START,
});

export const fetchInvestedStocksFail = (payload) => ({
  type: FETCH_INVESTED_STOCKS_FAIL,
  payload,
});

export const fetchInvestedStocksSuccess = (payload) => ({
  type: FETCH_INVESTED_STOCKS_SUCCESS,
  payload,
});

export const fetchInvestedStocks = () => async (dispatch, getState) => {
  try {
    const serverUrl = new URL(`${SERVER_URL}/investments/user/`);

    const headers = new Headers();
    const { accessToken } = getState().user;
    headers.append('authorization', `Bearer ${accessToken}`);

    dispatch(fetchInvestedStocksStart());
    const investmentsResult = await fetch(serverUrl, { headers });
    if (!investmentsResult.ok) {
      switch (investmentsResult.status) {
        case 401:
          throw new Error('Sorry, we cannot validate your identity. Please login and try again.');
        default:
          throw new Error('Oops, there\'s something wrong with our app.');
      }
    }

    const stocks = await investmentsResult.json();
    const symbols = stocks.map(({ symbol }) => symbol);
    dispatch(fetchInvestedStocksSuccess(
      symbols,
    ));
  } catch (err) {
    dispatch(fetchInvestedStocksFail(err));
  }
};
