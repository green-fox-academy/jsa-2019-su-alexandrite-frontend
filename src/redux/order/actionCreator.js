import { SERVER_URL } from 'react-native-dotenv';

import {
  PURCHASE_STOCK_SUCCESS,
  PURCHASE_STOCK_FAIL,
} from './actionType';

const purchaseStockSuccess = (payload) => ({
  type: PURCHASE_STOCK_SUCCESS,
  payload,
});

const purchaseStockFail = (payload) => ({
  type: PURCHASE_STOCK_FAIL,
  error: payload,
});

const purchaseStock = (symbol, shares, type, accessToken) => (dispatch) => {
  const orderUrl = new URL(`${SERVER_URL}/order`);
  fetch(orderUrl, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${accessToken}`,
    },
    body: JSON.stringify(
      {
        symbol,
        shares,
        type,
      },
    ),
  })
    .then((response) => {
      if (response.status !== 201) {
        switch (response.status) {
          case 401:
            throw new Error('Sorry, we cannot validate your identity. Please login and try again.');
          default:
            throw new Error('Oops! there\'s something wrong with our app.');
        }
      }
      return response.json();
    })
    .then(() => dispatch(purchaseStockSuccess()))
    .catch((error) => dispatch(purchaseStockFail(error.message)));
};

export default purchaseStock;
