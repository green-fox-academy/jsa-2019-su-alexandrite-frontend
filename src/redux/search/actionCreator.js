import { API_KEY, API_URL } from 'react-native-dotenv';

import { SEARCH_START, SEARCH_SUCCESS, SEARCH_FAILED } from './actionType';

const searchStart = () => ({
  type: SEARCH_START,
});

const searchSuccess = (payload) => ({
  type: SEARCH_SUCCESS,
  payload,
});

const searchFailed = (payload) => ({
  type: SEARCH_FAILED,
  payload,
});

const searchStockData = (input) => (dispatch) => {
  const url = new URL(`${API_URL}/search/${input}`);
  url.searchParams.append('token', API_KEY);
  dispatch(searchStart());
  fetch(url)
    .then((response) => {
      if (!response.ok) {
        switch (response.status) {
          case 404:
            throw new Error(`Unknown symbols were provided when you search for ${input}.`);
          case 400:
            throw new Error(`Invalid values were supplied when search for ${input}.`);
          default:
            throw new Error('Oops, there\'s something wrong with our app.');
        }
      }
      return response.json();
    })
    .then((response) => dispatch(searchSuccess(response)))
    .catch((err) => dispatch(searchFailed(err)));
};

export default {
  searchStockData,
};
