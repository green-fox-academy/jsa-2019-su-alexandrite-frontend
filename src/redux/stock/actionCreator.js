import { SERVER_URL, API_URL, API_KEY } from 'react-native-dotenv';

import {
  FETCH_STOCK_DETAILS_START,
  FETCH_STOCK_DETAILS_FAIL,
  FETCH_STOCK_DETAILS_SUCCESS,
  FETCH_HISTORY_CHART_DATA_START,
  FETCH_HISTORY_CHART_DATA_SUCCESS,
  FETCH_HISTORY_CHART_DATA_FAIL,
  RESET_STOCK_INFO,
  PLACE_ORDER_START,
  PLACE_ORDER_FAIL,
  PLACE_ORDER_SUCCESS,
  PURCHASE_STOCK_SUCCESS,
  PURCHASE_STOCK_FAIL,
} from './actionType';

import chartHelper from '../../common/chartHelper';

const fetchStockDetailsStart = () => ({
  type: FETCH_STOCK_DETAILS_START,
});

const fetchStockDetailsFail = (payload) => ({
  type: FETCH_STOCK_DETAILS_FAIL,
  payload,
});

const fetchStockDetailsSuccess = (payload) => ({
  type: FETCH_STOCK_DETAILS_SUCCESS,
  payload,
});

export const fetchStockDetails = (symbol) => (dispatch) => {
  const url = new URL(`${API_URL}/stock/${symbol}/advanced-stats`);
  url.searchParams.append('token', API_KEY);
  dispatch(fetchStockDetailsStart());
  fetch(url)
    .then((res) => {
      if (!res.ok) {
        switch (res.status) {
          case 404:
            throw new Error(`The stock ${symbol} you are looking for does not exist.`);
          default:
            throw new Error('Oops, there\'s something wrong with our app.');
        }
      }
      return res.json();
    })
    .then((res) => dispatch(fetchStockDetailsSuccess(res)))
    .catch((err) => dispatch(fetchStockDetailsFail(err)));
};

const fetchHistoryChartDataStart = () => ({
  type: FETCH_HISTORY_CHART_DATA_START,
});

const fetchHistoryChartDataFail = (payload) => ({
  type: FETCH_HISTORY_CHART_DATA_FAIL,
  payload,
});


const fetchHistoryChartDataSuccess = (histData, range) => {
  const payload = chartHelper.processChartData(histData, range);

  return {
    type: FETCH_HISTORY_CHART_DATA_SUCCESS,
    payload,
  };
};

export const fetchHistoryChartData = (symbol, range) => (dispatch) => {
  const url = new URL(`${API_URL}/stock/${symbol}/chart/${range}`);
  url.searchParams.append('token', API_KEY);
  url.searchParams.append('chartCloseOnly', true);
  url.searchParams.append('includeToday', true);
  dispatch(fetchHistoryChartDataStart());
  fetch(url)
    .then((res) => {
      if (!res.ok) {
        switch (res.status) {
          case 404:
            throw new Error(`The stock ${symbol} you are looking for does not exist.`);
          default:
            throw new Error('Oops, there\'s something wrong with our app.');
        }
      }
      return res.json();
    })
    .then((res) => {
      dispatch(fetchHistoryChartDataSuccess(res, range));
    })
    .catch((err) => dispatch(fetchHistoryChartDataFail(err)));
};

export const resetStockInfo = () => ({
  type: RESET_STOCK_INFO,
});

const placeOrderStart = () => ({
  type: PLACE_ORDER_START,
});

const placeOrderFail = (payload) => ({
  type: PLACE_ORDER_FAIL,
  payload,
});

const placeOrderSuccess = (payload) => ({
  type: PLACE_ORDER_SUCCESS,
  payload,
});

export const placeOrder = (symbol, shares, type) => async (dispatch, getState) => {
  const url = new URL(`${API_URL}/order`);
  const { accessToken } = getState((state) => state.user);
  url.searchParams.append('token', API_KEY);
  dispatch(placeOrderStart());
  const body = {
    symbol,
    shares,
    type,
  };
  const headers = new Headers();
  headers.append('authorization', `Bearer ${accessToken}`);
  fetch(url, { headers, body })
    .then((res) => {
      if (!res.ok) {
        switch (res.status) {
          case 404:
            throw new Error(`The stock ${symbol} you are looking for does not exist.`);
          default:
            throw new Error('Oops, there\'s something wrong with our app.');
        }
      }
      return res.json();
    })
    .then((res) => dispatch(placeOrderSuccess(res)))
    .catch((err) => dispatch(placeOrderFail(err)));
};

const purchaseStockSuccess = () => ({
  type: PURCHASE_STOCK_SUCCESS,
});

const purchaseStockFail = (payload) => ({
  type: PURCHASE_STOCK_FAIL,
  error: payload,
});

export const purchaseStock = (symbol, shares, type, accessToken) => (dispatch) => {
  const orderUrl = `${SERVER_URL}/order`;
  fetch(orderUrl, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${accessToken}`,
    },
    body: JSON.stringify(
      {
        symbol,
        shares,
        type,
      },
    ),
  })
    .then((response) => {
      if (response.status !== 201) {
        switch (response.status) {
          case 401:
            throw new Error('Sorry, we cannot validate your identity. Please login and try again.');
          default:
            throw new Error('Oops! there\'s something wrong with our app.');
        }
      }
      return response.json();
    })
    .then((response) => dispatch(purchaseStockSuccess(response.shares)))
    .catch((error) => dispatch(purchaseStockFail(error)));
};
