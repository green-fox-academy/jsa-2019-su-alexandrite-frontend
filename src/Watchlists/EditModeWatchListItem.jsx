import PropTypes from 'prop-types';
import React from 'react';
import {
  Text, View, TouchableHighlight,
} from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import styles from './styles';

export default function WatchListItem({
  isChecked,
  onSelect,
  ticker,
}) {
  return (
    <TouchableHighlight
      underlayColor="rgba(0,0,0,0.1)"
      onPress={onSelect}
    >
      <View style={styles.watchListItem}>
        <MaterialIcons
          style={[styles.checkButton, { color: isChecked ? '#666' : '#eee' }]}
          name={isChecked ? 'check-box' : 'check-box-outline-blank'}
        />
        <View style={styles.watchListItemColumn}>
          <Text style={[styles.watchListItemColumnText, { fontWeight: 'bold', color: '#566ed3' }]}>
            {ticker}
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  );
}

WatchListItem.propTypes = {
  ticker: PropTypes.string.isRequired,
  isChecked: PropTypes.bool.isRequired,
  onSelect: PropTypes.func.isRequired,
};
