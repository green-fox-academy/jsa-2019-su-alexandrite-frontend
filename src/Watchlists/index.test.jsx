import React from 'react';
import { shallow } from 'enzyme';
import { useDispatch, useSelector } from 'react-redux';
import renderer from 'react-test-renderer';
import Watchlists from './index';
import { fetchWatchlistDetails } from '../redux/watchList/actionCreator';

jest.useFakeTimers();

jest.mock('react-redux');
jest.mock('../redux/watchList/actionCreator');
jest.mock('../Watchlists/WatchList.jsx', () => '<Watchlist />');
jest.mock('../common/Popup', () => '<Popup />');
jest.mock('../common/ErrorMessage', () => '<ErrorMessage />');
jest.mock('../common/Card', () => '<Card />');

const SAMPLE_STATE = {
  watchlists: [
    {
      id: 123,
      name: 'Sample',
      stocks: [
        {
          ticker: 'AAPL', currPrice: 123, dailyChange: 1.3, volume: '12B',
        },
        {
          ticker: 'MSFT', currPrice: 123.1, dailyChange: -3.4, volume: '43M',
        },
      ],
    },
  ],
  watchlistDetailsError: null,
  isLoadingWatchlistDetails: false,
};

describe('<Watchlists />', () => {
  beforeEach(() => {
    useSelector.mockReturnValue(SAMPLE_STATE);
    useDispatch.mockReturnValue(jest.fn(() => { }));
  });

  afterAll(() => {
    jest.resetModules();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');

    const wrapper = shallow(<Watchlists />);

    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(
      <Watchlists />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should not dispatch fetch stock details if watchlists are empty', () => {
    const mockDispatch = jest.fn(() => { });
    useDispatch.mockReturnValueOnce(mockDispatch);
    fetchWatchlistDetails.mockReturnValueOnce('test_action');
    useSelector.mockReturnValueOnce({
      watchlists: [],
      watchlistDetailsError: null,
      isLoadingWatchlistDetails: false,
    });
    const wl = renderer.create(
      <Watchlists />,
    );
    wl.update(<Watchlists />);
    expect(mockDispatch).not.toHaveBeenCalled();
  });

  it('Should dispatch fetch stock details onload when watchlists not empty', () => {
    const mockDispatch = jest.fn(() => { });
    useDispatch.mockReturnValue(mockDispatch);
    fetchWatchlistDetails.mockReturnValueOnce('test_action');
    const wl = renderer.create(
      <Watchlists />,
    );
    wl.update(<Watchlists />);
    expect(useDispatch).toHaveBeenCalled();
    expect(mockDispatch).toHaveBeenCalledWith('test_action');
  });
});
