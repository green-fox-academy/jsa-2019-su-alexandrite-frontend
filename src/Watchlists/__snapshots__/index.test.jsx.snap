// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditFooter /> Should render and match the snapshot when in edit mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 6,
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#b40d0d",
          "fontSize": 12,
          "fontWeight": "bold",
          "textAlignVertical": "center",
        }
      }
    >
      <Text />
       0
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 6,
          "paddingHorizontal": 10,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#c12424",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
         Delete WatchList
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 6,
          "paddingHorizontal": 10,
          "paddingVertical": 8,
        }
      }
      testID="editorButton"
    >
      <Text
        style={
          Object {
            "color": "#666",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</View>
`;

exports[`<EditFooter /> Should render and match the snapshot when not in edit mode 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-end",
      "flex": 1,
      "marginTop": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 6,
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <Text />
  </View>
</View>
`;

exports[`<Watchlists /> Should render and match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={
      <<Card />>
        <<ErrorMessage />
          message="You don't have any watchlists right now.
Start by adding one! ;)"
        />
      </<Card />>
    }
    ListHeaderComponent={[Function]}
    contentContainerStyle={
      Object {
        "padding": 15,
        "paddingBottom": 60,
      }
    }
    data={
      Array [
        Object {
          "id": 123,
          "name": "Sample",
          "stocks": Array [
            Object {
              "currPrice": 123,
              "dailyChange": 1.3,
              "ticker": "AAPL",
              "volume": "12B",
            },
            Object {
              "currPrice": 123.1,
              "dailyChange": -3.4,
              "ticker": "MSFT",
              "volume": "43M",
            },
          ],
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <<Watchlist />
          item={
            Object {
              "id": 123,
              "name": "Sample",
              "stocks": Array [
                Object {
                  "currPrice": 123,
                  "dailyChange": 1.3,
                  "ticker": "AAPL",
                  "volume": "12B",
                },
                Object {
                  "currPrice": 123.1,
                  "dailyChange": -3.4,
                  "ticker": "MSFT",
                  "volume": "43M",
                },
              ],
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <<Popup />
    confirmButtonText="Add"
    confirmDisabled={true}
    onCancel={[Function]}
    onConfirm={[Function]}
    title="Create a watchlist"
    visible={false}
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Watchlist Name"
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomColor": "#eee",
          "borderBottomWidth": 1,
          "padding": 8,
        }
      }
      underlineColorAndroid="transparent"
    />
  </<Popup />>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Image
      source={1}
      style={null}
    />
  </View>
</View>
`;
